lib/realtime/tenants.ex

```diff
@@ -287,11 +292,62 @@ defmodule Realtime.Tenants do
     |> tap(fn _ -> Cache.distributed_invalidate_tenant_cache(tenant_id) end)
   end
 
+  @doc """
+  Sets last active at to a given tenant by external_id
+  """
+  @spec set_last_active_at(String.t()) :: Tenant.t() | nil
+  def set_last_active_at(external_id) do
+    external_id
+    |> get_tenant_by_external_id()
+    |> Tenant.last_active_at_changeset(%{last_active_at: NaiveDateTime.utc_now()})
+    |> Repo.update!()
+  end
+
+  @doc """
+  Returns list of active tenants as a Stream that are allocated to this node. The stream will also preload the extensions.
+
+  An active tenant is a tenant that has `last_active_at` higher than the given days.
+
+  Defaults to 5 days.
+  """
+  def stream_active_tenants(days \\ 5) do
+    region = Application.get_env(:realtime, :region)
+    regions = Nodes.region_to_tenant_regions(region)
+    region_nodes = Nodes.region_nodes(region)
+
+    from(t in Tenant,
```

Comment:
how should we handle new customers who have worked for more than five days but have not yet been terminated?

Reply:
updated to be in handle_continue